core/meta/src/TCling.cxx

```diff
@@ -1855,6 +1855,16 @@ static int HandleInterpreterException(cling::MetaProcessor* metaProcessor,
    return 0;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+
+void TCling::DiagnoseIfInterpreterException(std::exception &e) const
+{
+   cling::InterpreterException* ie = dynamic_cast<cling::InterpreterException*>(&e);
```

Comment:
Should be `if (cling::InterpreterException* ie = dynamic_cast<cling::InterpreterException*>(&e))` - the `dynamic_cast` might fail (return `nullptr`).